infer/inference.cc

```diff
@@ -162,6 +162,8 @@ unique_ptr<cfg::CFG> Inference::run(core::Context ctx, unique_ptr<cfg::CFG> cfg)
                         } else if (unreachableInstruction == nullptr) {
                             unreachableInstruction = expr.value.get();
                             locForUnreachable = expr.loc;
+                        } else {
+                            locForUnreachable = locForUnreachable.join(expr.loc);
```

Comment:
This change looks unrelated to reporting unused code via a tag, can you walk me through why we're doing this in this PR?

Reply:
The PR description briefly touches upon this, but prior to this change we were only using the `loc` for the first `Binding` within the dead basic block.

What this looks like right now is the following:

```ruby
if false
  method_call(10)
  #           ^^ error: dead code
  second_method_call(20)
end
```

`10` is the first `Binding` in the CFG, since we assign it to a temporary variable. We're now using this loc to dim the dead code, but dimming just that argument looks bad / buggy!

We've changed the logic to make the loc cover the _entire dead basic block_, which looks much better.

Note that further in this method, Sorbet explicitly does not emit an error on other dead blocks that follow from this block, since that would be a redundant error. We can't expand the loc to cover those, since they aren't guaranteed to be contiguous. So we still don't dim all dead code. We can address this limitation in a subsequent PR that adds a notion of "hint" errors that Sorbet doesn't emit on the command line but that LSP uses to dim the code.

Reply:
We should probably add a comment above this explaining what we are doing.
```suggestion
                        } else {
                            // Expand the loc to cover the entire dead basic block
                            locForUnreachable = locForUnreachable.join(expr.loc);
```